go/str_split_test.go

```diff
@@ -0,0 +1,84 @@
+package pehape
+
+import (
+	"testing"
+)
+
+func TestGetCapacity(t *testing.T) {
+	tests := []struct {
+		name     string
+		capacity int
+		mod      int
+		lenS     int
+		leng     int
+	}{
+		{"get mod 0", 10, 0, 100, 10},
+		{"get mod 1", 12, 1, 100, 9},
+		{"get mod 0", 1, 0, 10, 10},
+		{"get mod 1", 2, 1, 11, 10},
+		{"get mod 1", 2, 1, 11, 10},
+		{"get mod 2", 2, 2, 12, 10},
+		{"get mod 3", 2, 3, 13, 10},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			capacityAct, modAct := getCapacity(tt.lenS, tt.leng)
+			if capacityAct != tt.capacity || modAct != tt.mod {
+				t.Errorf("(%d, %d): expected (%d, %d), actual (%d, %d)", tt.lenS, tt.mod, tt.capacity, tt.mod, capacityAct, modAct)
+			}
+		})
+	}
+}
+
+func TestStrSplit(t *testing.T) {
+	tests := []struct {
+		name        string
+		expected    []string
+		errExpected error
+		s           string
+		length      []int
+	}{
+		{"with no pad \"hello\"", []string{"h", "e", "l", "l", "o"}, nil, "hello", []int{1}},
+		{"with pad 3 \"hello\"", []string{"hel", "lo"}, nil, "hello", []int{3}},
+		{"with pad 2 \"hello\"", []string{"he", "ll", "o"}, nil, "hello", []int{2}},
+		{"with pad 4 \"hello\"", []string{"hell", "o"}, nil, "hello", []int{4}},
+		{"with no pad \"Hello Friend\"", []string{"H", "e", "l", "l", "o", " ", "F", "r", "i", "e", "n", "d"}, nil, "Hello Friend", []int{1}},
+		{"with pad 5 \"Hello Friend\"", []string{"Hello", " Frie", "nd"}, nil, "Hello Friend", []int{5}},
+		{"with no data and pad", []string{""}, nil, "", []int{1}},
+		{"with length < 1", []string{}, errMustGreaterThanZero, "", []int{0}},
+		{"with 3 params", []string{}, errMust2Params, "", []int{0, 4}},
+		{"with no data with pad", []string{""}, nil, "", []int{5}},
+		{"with length > len(s)", []string{"hello"}, nil, "hello", []int{500}},
+	}
+	for _, tt := range tests {
+		tt := tt
```

Comment:
kenapa diginiin?

Reply:
Dari snippet vim-go 😁

Reply:
https://github.com/fatih/vim-go/blob/22b2273cfe562ac1c1af976ce77f18a3b1776f3c/gosnippets/UltiSnips/go.snippets#L456

Reply:
FYI ini karena ada t.Paralel(),
jd default snippet nya spt itu.
[source](https://gist.github.com/posener/92a55c4cd441fc5e5e85f27bca008721#how-to-solve-this)

Reply:
saya tambahin `t.Paralel()` biar ada gunanya wkwk

https://github.com/teknologi-umum/pehape/pull/40/commits/2df95fa7753bec41ba87c4dbe995a8b4754f336f#diff-479edf2b069d2035676e1c3f88582a43cb5fe7c66624761bcec3dd8185bbf4adR26